client/src/components/UnsavedChangesIndicator/UnsavedChangesIndicatorTimer.js

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import UnsavedChangesIndicator from './UnsavedChangesIndicator';
+
+/**
+ * A container component for UnsavedChangesIndicator that tracks the time since
+ * it was marked dirty, and updates the indicator level accordingly.
+ * By default it has a 5 minute notice level and 10 minute warning level.
+ */
+const UnsavedChangesIndicatorTimer = ({
+  isDirty = false,
+  minutes = {
+    notice: 5,
+    warning: 10,
+  },
+  UnsavedChangesIndicatorComponent = UnsavedChangesIndicator,
+}) => {
+  const [level, setLevel] = useState('none');
+  const [startTime, setStartTime] = useState(null);
+
+  /**
+   * Reset timer when isDirty changes
+   */
+  useEffect(() => {
+    if (isDirty) {
+      setStartTime(Date.now());
+      setLevel('none');
+    } else {
+      setStartTime(null);
+      setLevel('none');
+    }
+  }, [isDirty]);
+
+  /**
+   * Check elapsed time and update level accordingly
+   */
+  useEffect(() => {
+    if (!isDirty || startTime === null) {
+      return () => {};
+    }
+    const checkElapsedTime = () => {
+      const elapsed = Date.now() - startTime;
+      const warningMs = minutes.warning ? minutes.warning * 60000 : null;
+      const noticeMs = minutes.notice ? minutes.notice * 60000 : null;
+      if (warningMs && elapsed >= warningMs) {
+        setLevel('warning');
+      } else if (noticeMs && elapsed >= noticeMs) {
+        setLevel('notice');
+      } else {
+        setLevel('none');
+      }
+    };
+    const interval = setInterval(checkElapsedTime, 1000);
```

Comment:
I'm a bit confused about this interval. Aren't we checking more often than we need to?
We know what level we're on, and we know when we starting timing, and we know how long we need to wait between each level. Why don't we just set a timeout for that specific amount of time?

Reply:
This was honestly complex enough just using a interval. It could probably done with timeouts, but it would get confusing since you need to set two of them and like the intervals you'd need to keep cancelling them on onmount. It's just more to keep track of.

There's probably only a very minuscule performance gain from not using an interval, so keep as is